### ISSUE TITLE
plugin failed to load .netstandard2.0 library dll

### ISSUE THREAD

**Issue by TongZhai** [DotSpatial/DotSpatial#1327]
I am building a program using DotSpatial as frontend, I have the compiled binary runtime version: v4.0.30319 and Version: 2.0.0.465

The issue is with this plugin I am building that would need to use rdotnet DLL that is built on netstandard2.0, when I put the rdotnet DLL in my plugin's folder, the program (DemoMap) wouldn't launch and gives the following error message, then just quit:

`System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types.`

I verified by removing that DLL from the output folder, then it will launch. I've compiled my plugin on both 4.5 and 4.7.2 .net framework, same failure. This leads to the question can DotSpatial uses libraries built for netstandard?

**Comment by TongZhai** [comment]
Actually can DotSpatial be compiled for netstandard2.0 platform? it seems Microsoft is pushing for things to be on netstandard to be cross platform....

**Comment by jany-tenaj** [comment]
Try to compile DotSpatial on .Net 4.7.x or 4.8 and then include your plugin. I think DotSpatial is compiled against an older version of .Net that can't work with netstandard dlls.

**Comment by TongZhai** [comment]
I'll try that, is it feasible to compile DotSpatial on netstandard2.0 platform?

**Comment by jany-tenaj** [comment]
I haven't done anything with .net Standard yet. But I think it can't handle winforms, so you might have trouble compiling all of DotSpatial against .net Standard.

**Comment by TongZhai** [comment]
You know what's weird is that my previous application that is on .net framework 3.5 worked fine with this same netstandard library without problem. It appear to be the way DotSpatial is "loading" this library at the launching of the whole program that is failing, I wonder if that loading process can be examined per this issue... anyway, I will try recompile DotSpatial on higher version of .net framework...

**Comment by TongZhai** [comment]
I've tried recompiling dotspatial on 4.6.1 and 4.7.2, but both still failed to load that [netstandard 2.0 dll](https://respec-my.sharepoint.com/:u:/p/tong_zhai/EQenuRXEyjZEsY7WAq5ci0oBeKuqIG5boRIatdIVM_aqUA?e=YE7nrF). To reproduce the load problem, simply put this dll inside any of the plugins, for example the About plugin folder, then try to run DemoMap and see it fail. Please let me know if you can find a solution.

**Comment by TongZhai** [comment]
@jany-tenaj I was able to make the netstandard2.0 DLL work with the current 4.5.2 DotSpatial libraries! What I did was the following: 1) make the plugin itself to be .net 4.7.2 (it apparently can take in 4.5.2 DLLs); 2) trace the dependency of that netstandard2.0 DLL, it needs Dynamicinterop and Microsoft.Win32.Registry DLLs. So once these two were added into the plugin dependencies, it works now. So, I didn't have to compile DotSpatial to a higher .net framework version.
